pkg/endpoint/endpoint.go

```diff
@@ -1798,7 +1803,70 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 	return nil
 }
 
-// setPolicyRevision sets the policy revision with the given revision.
+// setPolicyRevision sets the policy wantedRev with the given revision.
 func (e *Endpoint) setPolicyRevision(rev uint64) {
 	e.policyRevision = rev
+	k := 0
+	for i, w := range e.policyRevisionSignals {
+		select {
+		case <-w.ctx.Done():
+			close(w.ch)
+		default:
+			if rev >= w.wantedRev {
+				close(w.ch)
+			} else {
+				// Make the closed policyRevisionSignals to go
+				// to the end of the slice so that we can keep them
+				// up afterwards
+				if i != k {
+					e.policyRevisionSignals[k] = w
```

Comment:
Why is this complicated logic required? Why don't we just `delete()` above?

Reply:
`policyRevisionSignals` is a slice, so no `delete()` for it.

Reply:
since order doesn't matter, we could use a `map[policySignal]bool` (or whatever value type).  I think policySignal can be a key (I hope)

Reply:
Now that I understand the logic, I would vote for a map or just making a second slice and putting the ones we want to keep in there. The sizes of these slices are so small that I'm not sure what we gain by trying to reuse the one slice.

Reply:
If you keep this logic, the comment just above wasn't clear. Perhaps:
 Make the closed policyRevisionSignals to go to the end of the slice so that we can keep them up afterwards
Move still-pending policyRevisionSignals entries to the start of the slice (overwriting closed, discardable ones). Then shorten the slice to only hold the pending entries.

Reply:
If we want to have deletion like a map does, but don't care about the value, then use a `map[policySignal]struct{}` which represents a set. This will make the code easier to understand.